server/app/grading_script.py

```diff
@@ -0,0 +1,56 @@
+import os
+import subprocess
+import json, urllib
+import httplib2
+import base64
+import ast
+import requests
+from oauth2client.tools import run
+from oauth2client.client import OAuth2WebServerFlow
+from oauth2client.file import Storage
+from apiclient.discovery import build
+from oauth2client.client import SignedJwtAssertionCredentials
+from oauth2client import gce
+
+http = httplib2.Http()
+email = '887076364996-opeevtug54963jn33liq618c1kjma8d5@developer.gserviceaccount.com'
+with open("ok-grader-for-CS169-project-9dd4b3bdce6e.p12") as f:
+        key = f.read()
+credentials = SignedJwtAssertionCredentials(email,
+                                            key,
+                                            'https://www.googleapis.com/auth/taskqueue')
+http = credentials.authorize(http)
+task_api = build('taskqueue', 'v1beta2', http=http)
+lease_req = task_api.tasks().lease(project='s~seventh-abacus-87719',
+                                    taskqueue='pull-queue',
+                                    leaseSecs=30,
+                                    numTasks=1)
+result = lease_req.execute()
+try:
+        task= result['items'][0]
+        print(task['id'])
+        payload = task['payloadBase64']
+        decodedResult = base64.urlsafe_b64decode(str(payload))
+        dictDecResult = ast.literal_eval(decodedResult)
+
+        subprocess.call(["chmod", "-R", "775", "grading"])
+        for key, value in dictDecResult.iteritems():
+                if key != "submit":
+                        f = open('grading/' + key, 'w')
+                        f.write(value.encode('utf-8'))
+                        f.close()
+        output = subprocess.check_output("./grading/grade.sh")
+        score_array = [int(p) for p in output.split() if p.isdigit()]
+
+        subm_id= task['tag']
+        req_url='https://seventh-abacus-87719.appspot.com/api/v1/grade/%s/add_grade'%subm_id
+        #access_token="ya29.longOkGoogleAuthToken&client_version=v1.3.30"
```

Comment:
I'm struggling with what's the right permissions model under the ok-server-spawns-a-docker-container model. Two possibilities:

(a) The ok server knows both auth & refresh tokens for an account that has course-level staff access, and it sends that to the container it spawns along with the student submission, just so the container can submit the score back. But whose credentials should it have? The person who pressed "start autograder" for the assignment? That's scary, because if I press that button, now the container has admin access and is executing untrusted code (bad).

(b) The ok server has a special account just for the autograder that's not allowed to do anything but post scores. Then, the `grader` value in a `Score` would be set to something reasonable. Is that a Google account, or do we have to invent a new permission system? Is there a reasonable place to store tokens for that account so that they can be passed to the autograder container?

Non solutions:

- give admin-level access to an autograder... the most permissions we can give it is course-level staff access.
- let anyone post scores without credentials